### PR TITLE
CLN: fix mypy warnings/errors

### PR DESCRIPTION
**Pull request by jbrockmendel**
When I run `mypy pandas` locally I get a handful of complaints.  This fixes about half of them.  The other half are about `FixedOffset` which IIRC we are handling elsewhere.

cc @simonjayhawkins

### CODE REVIEW

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -790,7 +790,7 @@ def _get_axes(group):
         return group.axes
 
 
-def _is_indexed_like(obj, axes):
+def _is_indexed_like(obj, axes) -> bool:
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py]: i think this edit is unrelated, snuck in on accident

**Comment by WillAyd** [comment]
What version of mypy are you using and what are the errors exactly? I'm very hesitant to add `# type: ignore` anywhere, especially if there's no urgency to do so

@angelaambroz fixed the issue with FixedOffset upstream in TypeShed and I think that will make its way into mypy 0.75

**Comment by jbrockmendel** [comment]
0.740

File: pandas/core/indexes/frozen.py [hunk 2]
```diff
@@ -68,7 +67,7 @@ def difference(self, other):
         return type(self)(temp)
 
     # TODO: Consider deprecating these in favor of `union` (xref gh-15506)
-    __add__ = __iadd__ = union
+    __add__ = __iadd__ = union  # type: ignore
```
Comment (WillAyd) [inline pandas/core/indexes/frozen.py]: The error now is essentially because union is unannotated, plus maybe a few more issues with subclass. In any case we should annotate instead of ignoring, especially since there's no rush on this one (current mypy in CI is still only 0.72)

Comment (jbrockmendel) [reply pandas/core/indexes/frozen.py]: sounds good, ill see if annotating union does the trick

Comment (jbrockmendel) [reply pandas/core/indexes/frozen.py]: looks like it worked for union, but not for _disabled.  removed the "type: ignore"s

**Comment by jreback** [comment]
thanks @jbrockmendel
